cmd/platform/export_test.go

```diff
@@ -1,30 +1,229 @@
+	if !regex.MatchString(err.Error()) {
+		t.Errorf("Platform Export command error message did not match expected regex\n\nerror message: '%v'\n\nregex pattern %s", err, regex.String())
+	}
+}
+
+// Test Platform Export Command --help, -h flag
+func TestPlatformExportCmd_HelpFlag(t *testing.T) {
+	err := testutils.ExecutePingctl("platform", "export", "--help")
+	if err != nil {
+		t.Errorf("Error executing platform export command: %v", err)
+	}
 
-	// Execute the command
-	err := rootCmd.Execute()
+	err = testutils.ExecutePingctl("platform", "export", "-h")
 	if err != nil {
-		testutils.PrintLogs(t)
-		t.Fatalf("Export Command failed. Make sure to have PingOne env variables set if test is failing.\nErr: %q, Captured StdOut: %s", err, stdout.String())
+		t.Errorf("Error executing platform export command: %v", err)
+	}
+}
+
+// Test Platform Export Command --service flag
+func TestPlatformExportCmd_ServiceFlag(t *testing.T) {
+	outputDir := os.TempDir() + "/pingctlTestPlatformExportServiceFlag"
+	err := testutils.ExecutePingctl("platform", "export", "--output-directory", outputDir, "--service", "pingone-protect", "--overwrite")
+	if err != nil {
+		t.Errorf("Error executing platform export command: %v", err)
+	}
+
+	// Empty output directory
+	err = os.RemoveAll(outputDir)
+	if err != nil {
+		t.Errorf("Error removing output directory: %v", err)
+	}
+}
+
+// Test Platform Export Command --service flag with invalid service
+func TestPlatformExportCmd_ServiceFlagInvalidService(t *testing.T) {
+	regex := regexp.MustCompile(`^invalid argument "invalid" for "--service" flag: unrecognized service 'invalid'\. Must be one of: [a-z-\s,]+$`)
+	err := testutils.ExecutePingctl("platform", "export", "--service", "invalid")
+
+	if !regex.MatchString(err.Error()) {
+		t.Errorf("Platform Export command error message did not match expected regex\n\nerror message: '%v'\n\nregex pattern %s", err, regex.String())
+	}
+}
+
+// Test Platform Export Command --export-format flag
+func TestPlatformExportCmd_ExportFormatFlag(t *testing.T) {
+	outputDir := os.TempDir() + "/pingctlTestPlatformExportExportFormatFlag"
+	err := testutils.ExecutePingctl("platform", "export", "--output-directory", outputDir, "--export-format", "HCL", "--overwrite", "--service", "pingone-protect")
+	if err != nil {
+		t.Errorf("Error executing platform export command: %v", err)
+	}
+
+	// Empty output directory
+	err = os.RemoveAll(outputDir)
+	if err != nil {
+		t.Errorf("Error removing output directory: %v", err)
+	}
+}
+
+// Test Platform Export Command --export-format flag with invalid forma
+func TestPlatformExportCmd_ExportFormatFlagInvalidFormat(t *testing.T) {
+	regex := regexp.MustCompile(`^invalid argument "invalid" for "--export-format" flag: unrecognized export format 'invalid'\. Must be one of: [A-Z]+$`)
+	err := testutils.ExecutePingctl("platform", "export", "--export-format", "invalid")
```

Comment:
You could update this utility function to take in t and an optional error regex to avoid the duplication

Reply:
added CheckExpectedError()